### PR TITLE
feat: added preliminary SDK support to "rand_data" plugin

### PR DESCRIPTION
**Pull request by patrickjcasey**
This pull request is the first pass at using the `hipcheck-sdk` crate to make a plugin for `hipcheck`

The goal of the SDK is to make it as easy as possible for plugin authors to write new plugins for hipcheck!

The hidden `hc plugin` command no accepts another flag `--sdk` that will use the `dummy_rand_data_sdk` plugin, rather than the `dummy_rand_data` plugin

There are some outstanding issues that still need to be addressed, whether in this issue, or in another issue in the future:
- How are we handling third-party dependencies that are exposed in public facing types (`serde_json::Value`, `schemars::schema::SchemaObject`, ...)
- How much documentation should be present in the `hipcheck-sdk` crate
- How do we test the `hipcheck-sdk` crate to ensure it is producing the desired behaviors and integrates cleanly with what `hipcheck` is expecting
- Does it make sense to add a `prelude` module so a user would be able to `use hipcheck_sdk::prelude::*` and have the most common imports?

### CODE REVIEW

**Comment by alilleybrinker** [comment]
Thanks @patrickjcasey! I'll take a look tomorrow! :smile:

File: hipcheck/src/plugin/types.rs
Comment (j-lanson) [inline hipcheck/src/plugin/types.rs:503]: delete or switch to `eprintln`

File: plugins/dummy_rand_data_sdk/src/main.rs
Comment (j-lanson) [inline plugins/dummy_rand_data_sdk/src/main.rs:47]: the error type saying `Response` when this is checking the type of the "request" key feels wrong

File: sdk/rust/src/plugin_server.rs
Comment (j-lanson) [inline sdk/rust/src/plugin_server.rs:67]: Remove todo comment

File: sdk/rust/src/lib.rs
Comment (j-lanson) [inline sdk/rust/src/lib.rs:27]: To answer your re-export question, I would just make a `pub mod export/deps { }` or something similar in this `lib.rs` file that declares all the types we need to re-export. I defer to Andrew if he has any opinions on this

File: sdk/rust/src/lib.rs
Comment (j-lanson) [inline sdk/rust/src/lib.rs:6]: If we don't need this to be public I don't think we want it to be

**Comment by j-lanson** [comment]
Re: a prelude crate

As far as I see, excluding the re-exports the only types users need access to is `QuerySchema, DynQuery, NamedQuery, Query, Plugin, PluginServer, PluginEngine`, plus the `error` module. And anyone using the SDK to write a plugin will need all of these. 

I'd be in favor of adding a `prelude` crate so that users that want to do so can simply import `*` as you have shown above.

Leave this and the reexport crate as a separate commit on top for now in case we need to walk it back

**Comment by j-lanson** [comment]
As for documentation and testing, those are important, but they can be done after the SDK is merged

File: sdk/rust/src/lib.rs
Comment (patrickjcasey) [reply sdk/rust/src/lib.rs:19]: for now I added a `deps` module that contains all of the types that need to be re-exported

**Comment by alilleybrinker** [comment]
I concur on the idea of a `prelude` module.

File: sdk/rust/src/lib.rs
Comment (patrickjcasey) [reply sdk/rust/src/lib.rs:11]: I added a prelude module, and updated the `dummy_rand_sdk` plugin to use the `use hipcheck_sdk::prelude::*` import

**Review by j-lanson — APPROVED** [review]
This looks good to me! Great work

**Pull request merged**
